ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteSessionImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.hibernate.ogm.datastore.ignite.impl;
+
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.event.spi.EventSource;
+import org.hibernate.internal.CriteriaImpl;
+import org.hibernate.ogm.OgmSessionFactory;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.OgmCriteriaLoader;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.impl.CriteriaCustomQuery;
+import org.hibernate.ogm.exception.NotSupportedException;
+import org.hibernate.ogm.hibernatecore.impl.OgmSessionImpl;
+import org.hibernate.resource.transaction.TransactionCoordinator;
+import org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl;
+
+public class IgniteSessionImpl extends OgmSessionImpl {
```

Comment:
> And I've always wanted to ask why don't you support criteria queries?

Lack of time and resources I think, we wanted to work on that but we decided to work on other features/dialects first.

This is also the reason it might take some time to review big PR like this one